packages/volto/src/components/manage/Widgets/SchemaWidget.jsx

```diff
@@ -142,160 +171,597 @@ const makeFieldsetList = (listOfFieldsets, intl) => {
-            },
-          };
-        case 'Floating-point number':
-        case 'label_float_field':
-        case 'Date/Time':
-        case 'label_datetime_field':
-        case 'Date':
-        case 'label_date_field':
-        case 'File':
-        case 'File Upload':
-        case 'Image':
-        case 'Yes/No':
-        case 'label_boolean_field':
-        case 'JSONField':
-        case 'Relation Choice':
-        case 'Relation List':
-          return {};
-        case 'Multiple Choice':
-        case 'label_multi_choice_field':
-        case 'Choice':
-        case 'label_choice_field':
-          return {
-            values: {
+      factory,
+    }),
+  });
+});
+
+config.registerUtility({
+  name: 'static_text',
+  type: 'fieldFactoryInitialData',
+  method: (intl) => ({
+    type: 'object',
+    widget: 'static_text',
+    factory: 'static_text',
+  }),
+});
+
+config.registerUtility({
+  name: 'hidden',
+  type: 'fieldFactoryInitialData',
+  method: (intl) => ({
+    type: 'string',
+    widget: 'hidden',
+    factory: 'hidden',
+  }),
+});
+
+config.registerUtility({
+  name: 'number',
+  type: 'fieldFactoryInitialData',
+  method: (intl) => ({
+    type: 'number',
+    factory: 'number',
+  }),
+});
+
+config.registerUtility({
+  name: 'radio_group',
+  type: 'fieldFactoryInitialData',
+  method: (intl) => ({
+    type: 'string',
+    choices: [],
+    widget: 'radio_group',
+    factory: 'radio_group',
+  }),
+});
+
+config.registerUtility({
+  name: 'checkbox_group',
+  type: 'fieldFactoryInitialData',
+  method: (intl) => ({
+    type: 'array',
+    widget: 'checkbox_group',
+    factory: 'checkbox_group',
+  }),
+});
+
+config.registerUtility({
+  name: 'textarea',
+  type: 'fieldFactoryInitialData',
+  method: (intl) => ({
+    type: 'string',
+    widget: 'textarea',
+    factory: 'textarea',
+  }),
+});
+
+/**
+ * schemaField used for modal form, when editing a field
+ * - based on the factory a set of fields is presented
+ * - fields can be moved to another fieldset
+ * @param {string} factory - the kind of field
+ * @param {Object} intl
+ * @param {*} fieldsets
+ * @param {Boolean} allowEditId
+ * @param {Object} extraFields
+ * @return {Object} - schema
+ */
+const schemaField = (
+  factory,
+  intl,
+  fieldsets,
+  allowEditId,
+  extraFields = {},
+) => {
+  const utility = config.getUtility({
+    name: factory,
+    type: 'fieldFactoryProperties',
+  });
+
+  const properties = utility.method
+    ? utility.method(intl)
+    : {
+        minLength: {
+          type: 'integer',
+          title: intl.formatMessage(messages.minLength),
+        },
+        maxLength: {
+          type: 'integer',
+          title: intl.formatMessage(messages.maxLength),
+        },
+        default: {
+          type: 'string',
+          title: intl.formatMessage(messages.defaultValue),
+        },
+        placeholder: {
+          type: 'string',
+          title: intl.formatMessage(messages.placeholder),
+        },
```

Comment:
I tried adding a placeholder to a text field in a content type schema, but it didn't show up when I edited the content type.

Reply:
It's a property of the field which can be used by the widget (or not). In this case the current widget in Plone doesn't use the value but when you use a custom widget (react-aria-components for example) you can use those properties.

Reply:
We can't just add properties to the schema editor that look useful but don't do anything. That's a recipe for frustration!

Reply:
See above comment.

Reply:
Both query param and placeholder are now opt-in.